src/gt4py/next/program_processors/runners/dace_fieldview/utility.py

```diff
@@ -0,0 +1,124 @@
+from gt4py.next.iterator.ir_utils import common_pattern_matcher as cpm
+from gt4py.next.program_processors.runners.dace_fieldview import gtir_python_codegen
+from gt4py.next.type_system import type_specifications as ts
+
+
+def as_dace_type(type_: ts.ScalarType) -> dace.typeclass:
+    """Converts GT4Py scalar type to corresponding DaCe type."""
+    match type_.kind:
+        case ts.ScalarKind.BOOL:
+            return dace.bool_
+        case ts.ScalarKind.INT32:
+            return dace.int32
+        case ts.ScalarKind.INT64:
+            return dace.int64
+        case ts.ScalarKind.FLOAT32:
+            return dace.float32
+        case ts.ScalarKind.FLOAT64:
+            return dace.float64
+        case _:
+            raise ValueError(f"Scalar type '{type_}' not supported.")
+
+
+def as_scalar_type(typestr: str) -> ts.ScalarType:
+    """Obtain GT4Py scalar type from generic numpy string representation."""
+    try:
+        kind = getattr(ts.ScalarKind, typestr.upper())
+    except AttributeError as ex:
+        raise ValueError(f"Data type {typestr} not supported.") from ex
+    return ts.ScalarType(kind)
+
+
+def debug_info(
+    node: gtir.Node, *, default: Optional[dace.dtypes.DebugInfo] = None
+) -> Optional[dace.dtypes.DebugInfo]:
+    location = node.location
+    if location:
+        return dace.dtypes.DebugInfo(
+            start_line=location.line,
+            start_column=location.column if location.column else 0,
+            end_line=location.end_line if location.end_line else -1,
+            end_column=location.end_column if location.end_column else 0,
+            filename=location.filename,
+        )
+    return default
+
+
+def filter_connectivities(offset_provider: Mapping[str, Any]) -> dict[str, gtx_common.Connectivity]:
+    """
+    Filter offset providers of type `Connectivity`.
+
+    In other words, filter out the cartesian offset providers.
+    Returns a new dictionary containing only `Connectivity` values.
+    """
+    return {
+        offset: table
+        for offset, table in offset_provider.items()
+        if isinstance(table, gtx_common.Connectivity)
+    }
+
+
+def get_domain(
+    node: gtir.Expr,
+) -> list[tuple[gtx_common.Dimension, dace.symbolic.SymbolicType, dace.symbolic.SymbolicType]]:
+    """
+    Specialized visit method for domain expressions.
+
+    Returns for each domain dimension the corresponding range.
+    """
+    assert cpm.is_call_to(node, ("cartesian_domain", "unstructured_domain"))
+
+    domain = []
+    for named_range in node.args:
+        assert cpm.is_call_to(named_range, "named_range")
+        assert len(named_range.args) == 3
+        axis = named_range.args[0]
+        assert isinstance(axis, gtir.AxisLiteral)
+        bounds = []
+        for arg in named_range.args[1:3]:
+            sym_str = gtir_python_codegen.get_source(arg)
+            sym_val = dace.symbolic.SymExpr(sym_str)
+            bounds.append(sym_val)
```

Comment:
Just an optional suggestion
```suggestion
        bounds = [
            dace.symbolic.SymExpr(gtir_python_codegen.get_source(arg))
            for arg in named_range.args[1:3]
        ]
```